packages/freezed/test/map_test.dart

```diff
@@ -298,5 +301,39 @@ void main() {
 
       expect(errorResult.errors, isNotEmpty);
     });
+    test('when is not marked @required', () async {
+      final main = await resolveSources(
+        {
+          'freezed|test/integration/main.dart': r'''
+library main;
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'multiple_constructors.dart';
+
+void main() {
+  final value = RequiredParams(a: 'a');
+
+  value.when((a) => a, second: (b) => b, third: (c) => c);
+}
+
+void redundant(@required String test) {
```

Comment:
That function seems unused. What is its purpose?

Reply:
It's actually just a sanity check to make sure `@required` flag is marked as redundant in that case..

https://github.com/rrousselGit/freezed/blob/79e00b1380103e5ba956d59c7b932a2ca7efc880/packages/freezed/test/map_test.dart#L325-L330

So it will fail if the linter rules change for whatever reason, otherwise it wouldn't be obvious that the test was useless..